sourcegen-generator-java/src/main/java/io/micronaut/sourcegen/JavaPoetSourceGenerator.java

```diff
@@ -581,6 +578,36 @@ private CodeBlock renderExpression(@Nullable ObjectDef objectDef, MethodDef meth
                 renderExpression(objectDef, methodDef, condition.right())
             );
         }
+        if (expressionDef instanceof ExpressionDef.And andExpressionDef) {
+            boolean isLeftCondition = andExpressionDef.left() instanceof ExpressionDef.Condition ||
+                andExpressionDef.left() instanceof ExpressionDef.And || andExpressionDef.left() instanceof ExpressionDef.Or;
+            boolean isRightCondition = andExpressionDef.right() instanceof ExpressionDef.Condition ||
+                andExpressionDef.right() instanceof ExpressionDef.And || andExpressionDef.right() instanceof ExpressionDef.Or;
+            return CodeBlock.concat(
+                (isLeftCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
+                renderExpression(objectDef, methodDef, andExpressionDef.left()),
+                (isLeftCondition) ? CodeBlock.of(")") : CodeBlock.of(""),
+                CodeBlock.of(" && "),
+                (isRightCondition) ? CodeBlock.of("(") : CodeBlock.of(""),
+                renderExpression(objectDef, methodDef, andExpressionDef.right()),
+                (isRightCondition) ? CodeBlock.of(")") : CodeBlock.of("")
+            );
```

Comment:
split this into multiple `if/else` to make this easier to read and/or consider rewriting to use switch expressions that assign the code block

Reply:
Wouldn't that be too many repeated lines? I am cool to change just thought it was too long for such small changes. 😄  

```
if (leftCondition && rightCondition) {
return CodeBlock.concat(
                CodeBlock.of("("),
                renderExpression(objectDef, methodDef, andExpressionDef.left()),
                CodeBlock.of(")"),
                CodeBlock.of(" && "),
                CodeBlock.of("("),
                renderExpression(objectDef, methodDef, andExpressionDef.right()),
                CodeBlock.of(")")
            );
} else if ( rightCondition) {
return CodeBlock.concat(
                renderExpression(objectDef, methodDef, andExpressionDef.left()),
                CodeBlock.of(" && "),
                CodeBlock.of("("),
                renderExpression(objectDef, methodDef, andExpressionDef.right()),
                CodeBlock.of(")")
            );
} else if (leftCondition) {
return CodeBlock.concat(
                CodeBlock.of("("),
                renderExpression(objectDef, methodDef, andExpressionDef.left()),
                CodeBlock.of(")"),
                CodeBlock.of(" && "),
                renderExpression(objectDef, methodDef, andExpressionDef.right()),
            );
} else {
return CodeBlock.concat(
                renderExpression(objectDef, methodDef, andExpressionDef.left()),
                CodeBlock.of(" && "),
                renderExpression(objectDef, methodDef, andExpressionDef.right()),
            );
} 
```